uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Form/ManageEntitlements.php

```diff
@@ -129,22 +129,33 @@ private function getContractsForCalculation(AbsencePeriod $absencePeriod) {
    */
   private function getActiveContractsForPeriod(AbsencePeriod $absencePeriod, $filter = []) {
     try {
-      $result = civicrm_api3('HRJobContract', 'getactivecontracts', [
-        'start_date' => $absencePeriod->start_date,
-        'end_date' => $absencePeriod->end_date,
-        'api.Contact.getvalue' => ['return' => 'display_name']
+      $contracts = CRM_Hrjobcontract_BAO_HRJobContract::getActiveContracts(
```

Comment:
Yep, but using that method the performance would be even poor, as I would have to call it for each contact. With getActiveContracts() I can get all the active contracts with a single query